### ISSUE TITLE
[Feature] Keep playlists updated

### ISSUE THREAD

**Issue by nlogozzo** [nlogozzo/NickvisionTubeConverter#681]
### Discussed in https://github.com/NickvisionApps/Parabolic/discussions/676

<div type='discussions-op-text'>

<sup>Originally posted by **ArthurOllier** November 18, 2023</sup>
as a user 
i want to be able to sync a folder with a playlist 
in order to not having to add a download myself pointing to a set directory

or

as a user 
i want to be able to save a download task (playlist in a set folder, as there is already a file by file history)
in order to play it again (on startup / on app start up ?)</div>

We will be going with this option:
> as a user
i want to be able to save a download task (playlist in a set folder, as there is already a file by file history)
in order to play it again (on startup / on app start up ?)

**Comment by nlogozzo** [comment]
@ArthurOllier Do you often rename the videos downloaded from the playlist?

I ask because I'm trying to think of how to check for videos already downloaded from the playlist. I'd like to compare the names of the files in the folder to the titles of the videos in the playlist and whatever is missing download them. However, if you rename the files, then it would think none of the videos are there and download them all again, in which we'd need another solution.

**Comment by LionelVallet** [comment]
> @ArthurOllier Do you often rename the videos downloaded from the playlist?
> 
> I ask because I'm trying to think of how to check for videos already downloaded from the playlist. I'd like to compare the names of the files in the folder to the titles of the videos in the playlist and whatever is missing download them. However, if you rename the files, then it would think none of the videos are there and download them all again, in which we'd need another solution.

You can store a hidden file containing the identifiers of successfully downloaded videos in the folder.
You may want to change file names, but also delete certain videos. In this case, we'd like to download only the new videos, and not re-download the deleted ones each time.

**Comment by nlogozzo** [comment]
> You can store a hidden file containing the identifiers of successfully downloaded videos in the folder.

That's a good idea! Thanks :)

**Comment by m1212e** [comment]
Hey, I'd really love this fature to become reality, is it realistic to help you out/contribute?

**Comment by ArthurOllier** [comment]
> [@ArthurOllier](https://github.com/ArthurOllier) Do you often rename the videos downloaded from the playlist?
> 
> I ask because I'm trying to think of how to check for videos already downloaded from the playlist. I'd like to compare the names of the files in the folder to the titles of the videos in the playlist and whatever is missing download them. However, if you rename the files, then it would think none of the videos are there and download them all again, in which we'd need another solution.

Sometimes but I'll go with @LionelVallet comment and opt for the hidden file if possible, thank you for working on this !

**Comment by LionelVallet** [comment]
Sometimes it’s the video uploaders who change the title as well, so matching must be done by ID and not by title.
The ID could be stored in the file metadata which allows server- or client-side renaming without the risk of duplicates. 

But another hidden file is necessary to handle intentional file deletion.

**Comment by nlogozzo** [comment]
Yes I think hidden file and saving by ID is the way I will implement this.

The next update, after 2025.6.0 I plan on fixing all the playlist issues: implementing this, and playlist quality support, etc...
